warrior/Framework/ClassUtils/gnmi_utils_class.py

```diff
@@ -0,0 +1,339 @@
+                    if "100%" in child.before+child.match.group(0):
+                        status = True
+                    break
+                if u_index == 3:
+                    testcase_Utils.pNote(child.before+child.match.group(0)+child.after)
+                    child.sendline('yes')
+                if u_index == 5:
+                    testcase_Utils.pNote(child.before+child.match.group(0))
+                    status = False
+                    break
+            except:
+                testcase_Utils.pNote("File Copy Failed", "error")
+                status = False
+                break
+        if status:
+            testcase_Utils.pNote("Client certificate copied Successfully.")
+        else:
+            testcase_Utils.pNote("Client certificate copy Failed.", "error")
+        return status
+
+    def execuate(self, cmd_string, uname, passwd, external_system=None,
+                 external_system_session=None, stop_after=None, gnmi_obj=None, script="No"):
+        """
+        Execute gNMI command using gNMI binary
+        :param cmd_string:
+        :param uname:
+        :param passwd:
+        :param external_system:
+        :param external_system_session:
+        :param stop_after:
+        :param gnmi_obj:
+        :param script:
+        :return:
+        """
+        status = False
+        result = None
+        excute = False
+        child = None
+        if external_system == None:
+            if gnmi_obj == None:
+                child = pexpect.spawn(cmd_string)
+                child.maxread = 50000
+            else:
+                child = gnmi_obj
+                child = pexpect.spawn(cmd_string)
+                child.maxread = 50000
+        else:
+            session_id = data_Utils.get_session_id(external_system, external_system_session)
+            child = data_Utils.get_object_from_datarepository(session_id)
+            child.sendline(cmd_string)
+        if script.lower() == "no":
+            u_index = child.expect(["username.*", ".*error.*", pexpect.EOF, pexpect.TIMEOUT],
+                                   timeout=5)
+            if u_index == 0:
+                child.sendline(uname)
+                status = True
+                testcase_Utils.pNote(child.match.group(0) + uname)
+                excute = True
+            else:
+                excute = False
+            p_index = child.expect(["password.*", ".*error.*", pexpect.EOF, pexpect.TIMEOUT],
+                                   timeout=5)
+            if p_index == 0:
+                testcase_Utils.pNote(child.match.group(0))
+                child.sendline(passwd)
+                excute = True
+            else:
+                excute = False
+        if excute or script.lower() == "yes":
+            if stop_after == None and "polling" not in cmd_string and "streaming" not in cmd_string:
+                j_index = child.expect(['.*(%|#|\$)', pexpect.EOF, pexpect.TIMEOUT], timeout=50)

```

Comment:
these prompts should not be hardcoded. It should come from the system datafile, supplied by the user.

Reply:
Addressed.